### PR TITLE
JEI Preview Fixes

### PR DESCRIPTION
**Pull request by Yefancy**
Fixed most of the issues mentioned by @ALongStringOfNumbers  in #157 . thanks for review

Fixes as follow:
- There is now a maximum zoom out limit
- Previews start at the maximum zoom level
- Right mouse button will also zoom structure preview **(for zoom in categories)**
- White highlighting over selected blocks
- The highlight layer sometimes renders incorrectly over the block
- Issues when scrolling through the preview JEI categories
- Block tooltips render as well as info when hovering over info icon
- Info icon needs to be updated
- Cannot zoom preview when scrolling through categories **(use right mouse button way)**
- Controllers are rendered full bright in the components section
- Layer not correctly set/reset when changing through multiblock tiers. 

Not covered:
- Weird seam on the corner of a block **(I can't reproduce it)**
- Assembly Line rotates about a fixed point not on multiblock

### CODE REVIEW

**Comment by ALongStringOfNumbers** [comment]
Since the RMB is intended as an alternative zoom method, would you mind updating the information panel with that? And also just including general instructions, like "click and drag to rotate" and "Use mousewheel to scroll"

One issue that Tech discovered was that for previews like the EBF with multiple versions (due to the coils) if you move to a preview other than cupronickel, and then scroll the mousewheel, the buttons to move through the tiers will be reset, but the actual preview will not be reset, which can result in something like this:
finalcoil

Where the preview is currently on the final coil tier, but has no option to go back through the tiers anymore. For any coil tier other than the final tier, this can be fixed by incrementing the coil tier display, then the back button will be restored, however for the final coil tier you have to close the entire preview.

Another minor issue (more of an annoyance) is that you can drag the preview while clicking on the buttons to move to the next tier coil. This means that the structure will snap back into place after the click to move to the next tier, which can be kinda jarring when quickly moving through the coil tiers. A good way of reproducing this is to quickly click through the tiers on the EBF preview, moving your mouse around within the box as you click.

As for the Seam, Tech was able to reproduce the one around the steam grinder, when he switched to using GUI scale large and not fullscreen (but just a maximized window)

**Comment by TechLord22** [comment]
I've noticed a very strange rendering effect on the EBF controller in the JEI preview. With Large GUI Scale, and a Maximized Windowed mode (not fullscreen mode), the following can be reproduced. I have tested with in Windowed Maximized mode also on auto GUI scale, and it did also appear, but it required a much steeper angle looking on the controller to appear (more on this later in the comment).

Open the JEI preview for the EBF. And look closely at the controller without adjusting the preview's rotation or zoom in any way. You should notice a black (and grayish) box behind the EBF controller's regular front overlay. As you zoom in, the black will eventually disappear, and zooming out will bring it back.

Here is an image of this black square from a distance.
Screenshot_20211001_220229

You can get closer to this black square by doing this: zooming in a little bit and holding down LClick and dragging down on the mouse a little bit. If you continue to repeat this process, you can get close enough to the black square to see it better. It only occurs at steep angles looking at the corner of the multiblock. Rotating the preview left or right makes it go away until the camera is returned to nearly the default rotation about the vertical axis. You can also see this effect by moving the mouse upwards instead of downwards in order to see the effect from below the multiblock structure.

I am very confident that the black square behind the EBF is actually the same texture as the Sifting Machine's top overlay if this helps narrow anything down at all.

Here is a closer image of this.
Screenshot_20211001_220548


You can do the same thing to get closer to the weird seam on the Steam Grinder, but I cannot tell what, if any, texture is behind the regular front overlay on the controller. 

I can conclude that the strange black square/sifter overlay happens on any multiblock using the texture that the EBF controller uses, even if they are all separately registered and have different resource locations. I cannot tell if the Fusion Reactors or Primitive Water Pump, multiblocks with unique controller faces, also has this happen. It is very hard to tell, but I believe the Steam Oven, Primitive Blast Furnace, Coke Oven, also all get that sifter texture to show as well.

**Comment by Yefancy** [comment]
> I've noticed a very strange rendering effect on the EBF controller in the JEI preview. With Large GUI Scale, and a Maximized Windowed mode (not fullscreen mode), the following can be reproduced. I have tested this also in Windowed Maximized mode also on Auto GUI scale, and it did also appear, but it required a much steeper angle looking on the controller to appear (more on this later in the comment).
> 
> Open the JEI preview for the EBF. And look closely at the controller without adjusting the preview's rotation or zoom in any way. You should notice a black (and grayish) box behind the EBF controller's regular front overlay. As you zoom in, the black will eventually disappear, and zooming out will bring it back.
> 
> Here is an image of this black square from a distance. Screenshot_20211001_220229
> 
> You can get closer to this black square by doing this: zooming in a little bit and holding down LClick and dragging down on the mouse a little bit. If you continue to repeat this process, you can get close enough to the black square to see it better. It only occurs at steep angles looking at the corner of the multiblock. Rotating the preview left or right makes it go away until the camera is returned to nearly the default rotation about the vertical axis. You can also see this effect from below as well by moving the mouse upwards instead of downwards.
> 
> I am very confident that the black square behind the EBF is actually the same texture as the Sifting Machine's top overlay if this helps narrow anything down at all.
> 
> Here is a closer image of this. Screenshot_20211001_220548
> 
> You can do the same thing to get closer to the weird seam on the Steam Grinder, but I cannot tell what, if any, texture is behind the regular front overlay on the controller.
> 
> I can conclude that the strange black square/sifter overlay happens on any multiblock using the texture that the EBF controller uses, even if they are all separately registered and have different resource locations. I cannot tell if the Fusion Reactors or Primitive Water Pump, multiblocks with unique controller faces, also has this happen. It is very hard to tell, but I believe the Steam Oven, Primitive Blast Furnace, and Coke Oven also all get that sifter texture to show as well.

Thanks, really detailed. I have successfully reproduced it after following your steps. So far, I don't know what the reason is.  I have some ideas, which will be tested when I have time. this problem isn't too worse for me now, ill fix another one first

File: src/main/resources/assets/gregtech/lang/en_us.lang [hunk 1]
```diff
@@ -3849,11 +3849,8 @@ gregtech.multiblock.universal.distinct.no=No
 gregtech.multiblock.universal.distinct.yes=Yes
 gregtech.multiblock.universal.distinct.info=If enabled, each Item Input Bus will be treated as fully distinct from each other for recipe lookup. Useful for things like Programmed Circuits, Extruder Shapes, etc.
 
-gregtech.multiblock.preview.tilt=Shift+LMB to tilt
-gregtech.multiblock.preview.zoom=Shift+RMB to zoom
-gregtech.multiblock.preview.pan=LMB+Drag to pan
-gregtech.multiblock.preview.move=RMB+Drag to move
-gregtech.multiblock.preview.reset=Scroll Mousewheel to reset
+gregtech.multiblock.preview.zoom=Use mousewheel or Shift+RMB to zoom
```
Comment (ALongStringOfNumbers) [inline src/main/resources/assets/gregtech/lang/en_us.lang:3852]: This should be updated to RMB+Drag instead of Shift+RMB

**Pull request merged**